index/scorch/segment/zap/dict.go

```diff
@@ -46,14 +47,14 @@ func (d *Dictionary) PostingsList(term []byte, except *roaring.Bitmap,
 }
 
 func (d *Dictionary) postingsList(term []byte, except *roaring.Bitmap, rv *PostingsList) (*PostingsList, error) {
-	if d.fst == nil {
+	if d.fstReader == nil {
 		if rv == nil || rv == emptyPostingsList {
 			return emptyPostingsList, nil
 		}
 		return d.postingsListInit(rv, except), nil
 	}
 
-	postingsOffset, exists, err := d.fst.Get(term)
+	postingsOffset, exists, err := d.fstReader.Get(term)
```

Comment:
Trying to load this back into my head...

And, looks like there are two separate, independent changes here...

1) add a reused fstReader to the Dictionary

2) fix to the IndexSnapshot.fieldDicts reuse

The first change of the Dictionary.fstReader looks pretty straightforward and looks to good to me.  The 2nd change, though, is taking a bit more mental effort on my part and still thinking about it.  So, one thought is perhaps splitting this PR into two independent PR's, if you want that 1st Dictionary.fstReader change to get in sooner.

Reply:
@steveyen The 1st change that you speak about is what caused the panic seen in Alex's tests, and we had to revert it with this commit: https://github.com/blevesearch/bleve/commit/477688c147c679e43f471d36286c1e462295554b

To avoid that panic, these set of changes would all need to remain together.